ocaml/xapi/xapi_cluster.ml

```diff
@@ -50,44 +53,48 @@ let create ~__context ~pIF ~cluster_stack ~pool_auto_join ~token_timeout
       let host = Helpers.get_master ~__context in
       let pifrec = Db.PIF.get_record ~__context ~self:pIF in
       assert_pif_prerequisites (pIF, pifrec) ;
+      let open Cluster_interface in
       let ip_addr = ip_of_pif (pIF, pifrec) in
       let hostuuid = Inventory.lookup Inventory._installation_uuid in
       let hostname = Db.Host.get_hostname ~__context ~self:host in
       let member =
         if Xapi_cluster_helpers.cluster_health_enabled ~__context then
-          Cluster_interface.(
-            Extended
-              {
-                ip= Ipaddr.of_string_exn (ipstr_of_address ip_addr)
-              ; hostuuid
-              ; hostname
-              }
-          )
+          Extended
+            {
+              ip= Ipaddr.of_string_exn (ipstr_of_address ip_addr)
+            ; hostuuid
+            ; hostname
+            }
         else
-          Cluster_interface.(IPv4 (ipstr_of_address ip_addr))
+          IPv4 (ipstr_of_address ip_addr)
       in
       let token_timeout_ms = Int64.of_float (token_timeout *. 1000.0) in
       let token_timeout_coefficient_ms =
         Int64.of_float (token_timeout_coefficient *. 1000.0)
       in
       let init_config =
         {
-          Cluster_interface.member
+          member
         ; token_timeout_ms= Some token_timeout_ms
         ; token_coefficient_ms= Some token_timeout_coefficient_ms
         ; name= None
+        ; cluster_stack=
+            Cluster_stack.of_version (cluster_stack, cluster_stack_version)
         }
       in
       Xapi_clustering.Daemon.enable ~__context ;
+      maybe_switch_cluster_stack_version ~__context ~self:cluster_host_ref
+        ~cluster_stack:(cluster_stack ^ Int64.to_string cluster_stack_version) ;
```

Comment:
It would be nicer to use `Cluster_stack.t` rather than all those string concatenations here and below.